app/models/job.rb

```diff
@@ -38,20 +38,30 @@ class << self
   STATES.each do |s|
     define_method("#{s.downcase}?") { self.state == s }
   end
+
+  # TODO: Check if the complexity in the complete? method is required
```

Comment:
Why would this not be required?  What would the alternative be?  If we can simplify this by setting the state correctly in event processor, let's do so as part of this PR.

Reply:
Looking through the code base the `complete?` method isn't called anywhere. Overriding the `complete?` method but not `failed?`, `cancelled?`, etc.. looks a bit strange IMO.

Whilst we have the opportunity I have removed the override: https://github.com/openflighthpc/flight-scheduler-controller/pull/12/commits/87ac9ae0e46968fc7f855449e54fdd5e3d3901ad
Instead the `complete?` method is a short hand to `job.state == 'COMPLETED'`

Whilst I agree setting the `state` could use a simplification any solution would require architectural changes. This is best left for a future PR.

Reply:
As discussed, can we update `task.array_job` in `node_failed_task` and `node_completed_task` too.